lib/mayaUsd/commands/baseExportCommand.cpp

```diff
@@ -265,30 +265,13 @@ try
         frameSamples.insert(tmpArgList.asDouble(0));
     }
 
-    // Get the objects to export as a MSelectionList
     MSelectionList objSelList;
-    if (argData.isFlagSet(kSelectionFlag)) {
-        MGlobal::getActiveSelectionList(objSelList);
-    }
-    else {
-        argData.getObjects(objSelList);
-
-        // If no objects specified, then get all objects at DAG root
-        if (objSelList.isEmpty()) {
-            objSelList.add("|*", true);
-        }
-    }
-
-    // Convert selection list to jobArgs dagPaths
     UsdMayaUtil::MDagPathSet dagPaths;
-    for (unsigned int i=0; i < objSelList.length(); i++) {
-        MDagPath dagPath;
-        status = objSelList.getDagPath(i, dagPath);
-        if (status == MS::kSuccess)
-        {
-            dagPaths.emplace(dagPath);
-        }
+    bool exportSelected = argData.isFlagSet(kSelectionFlag);
+    if (!exportSelected) {
+        argData.getObjects(objSelList);
     }
+    GetFilteredSelectionToExport(exportSelected, objSelList, dagPaths);
```

Comment:
This bit where we get an object list and add to a set for export has been moved to util.h/.cpp as a utility function shared with the translator.  The one thing that the command can do differently is get a list of objects explicitly passed to the command to work on.